src/crimson/osd/pg_backend.cc

```diff
@@ -568,6 +568,54 @@ seastar::future<> PGBackend::omap_get_keys(
   //ctx->delta_stats.num_rd++;
 }
 
+seastar::future<> PGBackend::omap_get_vals(
+  const ObjectState& os,
+  OSDOp& osd_op) const
+{
+  std::string start_after;
+  uint64_t max_return;
+  std::string filter_prefix;
+  try {
+    auto p = osd_op.indata.cbegin();
+    decode(start_after, p);
+    decode(max_return, p);
+    decode(filter_prefix, p);
+  } catch (buffer::error&) {
+    throw ceph::osd::invalid_argument{};
+  }
+
+  max_return = \
+    std::min(max_return, local_conf()->osd_max_omap_entries_per_request);
+
+  // TODO: truly chunk the reading
+  return maybe_get_omap_vals(store, coll, os.oi, start_after).then(
+    [=, &osd_op] (bool, ceph::os::FuturizedStore::omap_values_t vals) {
```

Comment:
i think we should handle `bool done` as well. currently, CyanStore always return `<true, values>` even if the size of `values` is truncated by the `MAX_KEYS_PER_OMAP_GET_CALL`. but in future, we could but if we should repeat until `done` or `result.length() >= local_conf()->osd_max_omap_bytes_per_request`?

if we assume that the store will always return the full set of values whose key is greater or equal to start, we should `assert(done)` here.

Reply:
Put `asser(done)` for now.